src/main/java/com/stripe/net/HttpURLConnectionClient.java

```diff
@@ -78,18 +79,13 @@ public StripeResponse request(StripeRequest request) throws ApiConnectionExcepti
     }
   }
 
-  static Map<String, String> getHeaders(StripeRequest request) {
-    Map<String, String> headers = new HashMap<>();
+  static HttpHeaders getHeaders(StripeRequest request) {
+    Map<String, List<String>> headerMap = new HashMap<>(request.headers().map());
 
-    headers.putAll(request.headers());
+    headerMap.put("User-Agent", Arrays.asList(buildUserAgentString()));
+    headerMap.put("X-Stripe-Client-User-Agent", Arrays.asList(buildXStripeClientUserAgentString()));
```

Comment:
Just calling out that it strikes me as a little weird that you can't use `HttpHeaders` to build a set of headers, and instead have to manipulate a `Map` and then create `HttpHeaders`.

I think based on your PR description that it's supposed to be a safety feature to prevent reuse of an `HttpHeaders` object, but the limitation feels a little over-restrictive IMO.

Reply:
Yeah, I was on the fence about it too. I'll add a helper to `HttpHeaders`.